src/storage/src/table/state_table.rs

```diff
@@ -168,83 +169,114 @@ impl<'a, S: StateStore> StateTableRowIter<'a, S> {
     }
 
     pub async fn next(&mut self) -> StorageResult<Option<Row>> {
-        let next_flag;
+        let mut next_flag;
         let mut res = None;
```

Comment:
Also better put this inside the loop, remove `mut` and `= None`, so that the compiler will ensure `res` will be assigned only once.